### ISSUE TITLE
storage,tsdb: incorrect graphs for storage byte volumes

### ISSUE THREAD

**Issue by jbowens** [cockroachdb/cockroach#105923]
I believe the DB Console Storage dashboard is presenting storage metrics incorrectly. It appears to downsample using averaging (maybe a consequence of these being modeled as gauges instead of counters within the Cockroach metric system).

From the Storage dashboard:
<img width="990" alt="Screenshot 2023-06-30 at 2 53 31 PM" src="https://github.com/cockroachdb/cockroach/assets/867352/b8d62f8f-d916-4e60-8513-e9d5f0cfb997">

Using sum downsampling:
<img width="985" alt="Screenshot 2023-06-30 at 2 53 40 PM" src="https://github.com/cockroachdb/cockroach/assets/867352/9be45a13-e538-4068-bdce-67afe733b5a4">

Related to #102729, #75523, #99922.

Jira issue: CRDB-29274

**Comment by jbowens** [comment]
I'm second-guessing myself here. I'll run an experiment again this week and write up something definitive.

**Comment by jbowens** [comment]
I suspect the unit should just be MiB/s, because these values are scaled to be per-second. Can someone from @cockroachdb/cluster-observability confirm?

**Comment by jbowens** [comment]
> I'd expect area under the curve for some interval to give me a value in bytes.

Yeah, the current ambiguous element is what is the unit of time. Previously my expectation was that if a point represented a 10-second span, its value represented the cumulative bytes flushed (or compacted, ingested, etc) over that 10 second time period. To sum the area under the curve, all you need to do is add up all the points. But that does not appear to be what's shown.

Do you know where this value gets average to a per-second rate?

**Comment by jbowens** [comment]
Excluding the `nonNegativeRate` would produce a monotonically increasing value of the total bytes flushed since the process began. We want a rate, we just want the rate to have some known time interval. It's unclear to me what time interval we're seeing now, and if it is scaled to per-second or not.

**Comment by nicktrav** [comment]
Questions for obs-inf: what are the units.

**Comment by ericharmeling** [comment]
> We want a rate, we just want the rate to have some known time interval. It's unclear to me what time interval we're seeing now, and if it is scaled to per-second or not.

Apologies for the delay on this!

> It appears to downsample using averaging

Currently, the frontend constructs a query that uses the MAX downsampler if none is explicitly provided: https://github.com/cockroachdb/cockroach/blob/master/pkg/ui/workspaces/db-console/src/views/shared/containers/metricDataProvider/index.tsx#L56

I think we want to reconsider that default behavior: https://github.com/cockroachdb/cockroach/issues/106215.

> It's unclear to me what time interval we're seeing now, and if it is scaled to per-second or not.

For graph time intervals under 30 minutes, I believe that we use a 10s resolution. And I think `nonNegativeRate` provides the rate of change over whatever the resolution time, in terms of individual, downsampled observations: https://github.com/cockroachdb/cockroach/blob/master/pkg/ts/query.go#L425. So while it's technically not a recorded per-second rate of change, it can be expressed as "per second", as it's a first-order derivative between two observations over the sample period.

**Comment by jbowens** [comment]
Thanks @ericharmeling! Sorry, I'm still a bit mixed up here.

Say we use a 10s resolution, and we have a data point X1 at time T and a data point X2 at T+10s. With the `nonNegativeRate` I was expecting to see the point at T+10s to be X2-X1, the delta between the two observations.

Looking at the code you linked, it looks like we compute X2-X1, then we divide by the difference between the timestamps, scaled to the `samplePeriod`. Is the `samplePeriod` 1 second here, so the rate of changed is scaled to per-second?

**Comment by ericharmeling** [comment]
> Looking at the code you linked, it looks like we compute X2-X1, then we divide by the difference between the timestamps, scaled to the samplePeriod. Is the samplePeriod 1 second here, so the rate of changed is scaled to per-second?

Ahh - okay. I think I misinterpreted the calculation of `rateOfChange`. And I was reading "X2-X1/samplePeriod". But that's incorrect. `samplePeriod` is 10s, which should be the same as the difference between time T and T+10. So, the unit of change between the observations should be "per 10 seconds" _as calculated by rateOfChange_.  I believe the sample period scaling is then happening on the output of `rateOfChange_` here: https://github.com/cockroachdb/cockroach/blob/2ec2b61627cdadc8b269e85d6b4a64c669b6c140/pkg/ts/query.go#L709 

The UI values are per-second though.

**Comment by jbowens** [comment]
Thanks @ericharmeling! I'll update the DB console labels.
